debug/local-testing-debugging.md

```diff
@@ -0,0 +1,198 @@
+# Automated Settings for Running and Debugging Knative Jobs Locally on ppc64le
+
+This document helps set up a local (non-Prow) environment for testing or debugging Knative projects on a ppc64le VM. 
+
+The following Knative projects are currently supported for local testing and debugging:
+
+- operator
+- client
+- eventing
+
+## Prerequisites  
+
+- **`ppc64le` VM** with KinD and `docker/podman` installed.
+- Python (3+) installed on the VM
+- **Required Secrets:** 
+  - `config.json`: Required for cluster access to IBM Container Registry.  
+
+---
+
+## Usage
+
+### 1. Enter into the debug directory
+
+```bash
+cd debug
+```
+
+### 2. Build test image
+
+Build testing image `quay.io/p_serverless/knative-prow-tests:debug` required for testing environment
+
+```bash
+docker buildx build --platform linux/ppc64le -t quay.io/p_serverless/knative-prow-tests:debug --load -f Dockerfile .
+```
+
+### 3. Add IBM container registry keys
+
+Replace/Add creds of the IBM container registry in the config.json file
+
+```bash
+{
+  "auths": {
+    "na.artifactory.swg-devops.com": {
+      "auth": "<add-auth-key-here>"
+    },
+    "quay.io": {
+      "auth": "<add-auth-key-here>"
+    },
+    "icr.io": {
+      "auth": "<add-auth-key-here>"
+    }
+  }
+}
+```
+
+### 4. Edit `.env`. 
+
+Update the Knative repo project name, branch and other environment variable required to run the test. To update the variable values refer `ppc64le` prow  Knative job configs: [knative-prow-jobs](https://github.com/kabhiibm/test-infra/tree/master/config/jobs/periodic) 
+
+For example: `.env` file for `operator` testing
+
+```bash
+KIND_IMAGE=quay.io/powercloud/kind-node
+K8S_VERSION=v1.34.1
+USE_DOCKER=true
+export DEBUG=true
+
+# Test repo related variables
+export KNATIVE_ORG=knative
+export KNATIVE_REPO=operator
+export KNATIVE_RELEASE=main
+
+# Common test variables
+export KO_FLAGS='--platform=linux/ppc64le'
+export PLATFORM=linux/ppc64le
+export KO_DOCKER_REPO=icr.io/upstream-k8s-registry/knative
+export DOCKER_CONFIG=/root/.docker
+
+# Test config variables
+export INGRESS_CLASS=contour.ingress.networking.knative.dev
+#export SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt
+#export SYSTEM_NAMESPACE=knative-eventing
+#export SCALE_CHAOSDUCK_TO_ZERO=1
+
+# Eventing KAFKA Broker
+#export EVENTING_KAFKA_BROKER_CHANNEL_AUTH_SCENARIO=SASL_SSL
+#export EVENTING_NAMESPACE=knative-eventing
+#export BROKER_CLASS=Kafka
+```
+
+### 5. Start testing/debugging
+
+Run `debug-knative.py` program to setup the KinD (1 master, 1 worker nodes). It will automatically run the admustment scripts for debugging. It will also create and provide a container with testing/debugging environment.
+
+```bash
+$ ./debug-knative.py 
```

Comment:
Send `KNATIVE_ORG`, `KNATIVE_REPO` and `KNATIVE_RELEASE` as input parameters to debug-knative.py file.  That way user doesn't have to modify the env file everytime